config.go

```diff
@@ -108,9 +123,46 @@ func Parse(cfg map[string]string) (Config, error) {
 	if err != nil {
 		return Config{}, fmt.Errorf("invalid TLS config: %w", err)
 	}
+	err = setSASLConfigs(&parsed, cfg)
+	if err != nil {
+		return Config{}, fmt.Errorf("invalid SASL config: %w", err)
+	}
 	return parsed, nil
 }
 
+func setSASLConfigs(parsed *Config, cfg map[string]string) error {
```

Comment:
Off-topic, WDYT about making `setSASLConfigs` a method in `Config`?

Reply:
> on the other hand it doesn't care about those params if SASL is explicitly disabled.

Here I had the scenario in mind where you want to temporarily disable a setting without removing the params themselves because you will want to enable it again later. Idk, might be a foot gun, it is explicit though 🤷 I imagine we could log a warning in that case, saying that sasl params were found but it is explicitly disabled.

> WDYT about making `setSASLConfigs` a method in `Config`?

If we do this it will create a strange situation where we have `setSASLConfigs` on `Config` and other parsing functions would still be independent. I think config parsing is becoming huge so it's again the package global functions problem that we're trying to solve, right? If that's the case, I propose attaching everything regarding config parsing including `Parse` to `Config` and let the struct itself handle how it parses config options (similar to how a struct can implement [`json.Unmarshaler`](https://pkg.go.dev/encoding/json#Unmarshaler) to define how it parses JSON).

```
func (c *Config) Parse(cfg map[string]string) error {
	// ...
}

// Parsing the config would look like this
func (s *Source) Configure(ctx context.Context, cfg map[string]string) error {
	err := s.Config.Parse(cfg)
	if err != nil {
		return fmt.Errorf("config is invalid: %w", err)
	}
	return nil
}
```

This sounds like a bigger refactoring though, so if you like the idea feel free to create an issue and tackle it separately.

Reply:
> Here I had the scenario in mind where you want to temporarily disable a setting without removing the params themselves because you will want to enable it again later. Idk, might be a foot gun, it is explicit though 🤷 I imagine we could log a warning in that case, saying that sasl params were found but it is explicitly disabled.

That's a good away to approach it. However, if we want to make it possible to easily, temporarily disable a setting, then I think we might be solving the wrong problem. Whoever's sending the configuration should keep it clean: when needed -- add it, when not -- remove. 

> If we do this it will create a strange situation where we have `setSASLConfigs` on `Config` and other parsing functions would still be independent.

For sure, we'd do the same for other functions.

> This sounds like a bigger refactoring though, so if you like the idea feel free to create an issue and tackle it separately.

I agree that the config's parsing became relatively big and close to a point where we should refactor it. IMHO it's not too big at the moment, but if we add more params, it will probably become so. I like the idea of refactoring, so I'll create an issue.

Reply:
https://github.com/ConduitIO/conduit-connector-kafka/issues/13